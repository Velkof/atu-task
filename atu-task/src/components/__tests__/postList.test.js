import React from 'react';
import renderer from 'react-test-renderer';
import PostList from '../postList';

const props = {
    loading: false,
    posts: [{
        id: 1,
        title: 'Post title',
        userId: 2,
        body: 'Post body'
    },
    {
        id: 2,
        title: 'Post title 2',
        userId: 3,
        body: 'Post body 2'
    }],
    getPosts: () => {}
}

test('<PostList /> renders successfully', () => {
  const tree = renderer.create(
    <PostList
        {...props}
    />
  ).toJSON();
  
  expect(tree).toMatchSnapshot();
});
