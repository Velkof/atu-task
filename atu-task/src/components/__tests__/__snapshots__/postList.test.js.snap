// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostList /> renders successfully 1`] = `
<div
  className="post-list"
>
  <h1>
    Posts List
  </h1>
  <div
    className="post"
  >
    <div
      className="post-id"
    >
      <p>
        1
      </p>
    </div>
    <div
      className="info-container"
    >
      <p
        className="title"
      >
        Post title
      </p>
    </div>
    <div
      className="button-container"
    >
      <button
        onClick={[Function]}
      >
        Read post
      </button>
    </div>
  </div>
  <div
    className="post"
  >
    <div
      className="post-id"
    >
      <p>
        2
      </p>
    </div>
    <div
      className="info-container"
    >
      <p
        className="title"
      >
        Post title 2
      </p>
    </div>
    <div
      className="button-container"
    >
      <button
        onClick={[Function]}
      >
        Read post
      </button>
    </div>
  </div>
</div>
`;
